CoreWebServer/src/au/gov/asd/tac/constellation/webserver/help/SphinxHelpDisplayer.java

```diff
@@ -0,0 +1,322 @@
+        final BufferedReader reader = new BufferedReader(new InputStreamReader(new ByteArrayInputStream(buf.toByteArray()), StandardCharsets.UTF_8));
+        return reader.lines().collect(Collectors.toList());
+    }
+
+    private static List<String> initHelp() throws IOException {
+        helpSource = System.getProperty(CONSTELLATION_HELP);
+        LOGGER.log(Level.INFO, "Help source: {0}", helpSource);
+
+        if (helpSource == null) {
+            LOGGER.info(String.format("help_map file at zip resource %s, %s", HELP_ZIP, HELP_MAP));
+            return getResourceZipFile(HELP_MAP);
+        } else if (helpSource.startsWith("http")) {
+            final String url = String.format("%s/%s", helpSource, HELP_MAP);
+            LOGGER.log(Level.INFO, "help_map file at {0}", url);
+            return getHttpFile(url);
+        } else {
+            LOGGER.log(Level.INFO, "help_map file at zip {0}, {1}", new Object[]{helpSource, HELP_MAP});
+            return getZipFile(helpSource, HELP_MAP);
+        }
+    }
+
+    /**
+     * Return the help map mapping helpIds to documentation page paths.
+     * <p>
+     * This can be in any one of three places:
+     * <ul>
+     * <li>in a zip file in modules/resources/ext</li>
+     * <li>in a named external zip file</li>
+     * <li>at a web location</li>
+     * </ul>
+     *
+     * @return The helpId to documentation path mapping.
+     */
+    private static synchronized Map<String, String> getHelpMap() {
+        if (helpMap == null) {
+            try {
+                final List<String> lines = initHelp();
+                helpMap = new HashMap<>();
+                lines.forEach(line -> {
+                    final int ix = line.indexOf(',');
+                    final String helpId = line.substring(0, ix).strip();
+                    if (!helpId.isEmpty() && !helpId.startsWith("#")) {
+                        final String helpPath = line.substring(ix + 1);
+                        helpMap.put(helpId, helpPath);
+                    }
+                });
+            } catch (final IOException ex) {
+                LOGGER.log(Level.INFO, "Fetching help map:", ex);
+
+                // If we couldn't read the file the first time,
+                // it won't magically work the next time, so stop trying.
+                //
+                helpMap = Map.of();
+            }
+        }
+
+        return helpMap;
+    }
+
+    public static void copy(final String filepath, final OutputStream out) throws IOException {
+        final String p = filepath.startsWith("/") ? filepath.substring(1) : filepath;
+        if (helpSource == null) {
+            copyFromZipResource(p, out);
+        } else {
+            copyFromZipFile(helpSource, p, out);
+        }
+    }
+
+    @Override
+    public boolean display(final HelpCtx helpCtx) {
+        final String helpId = helpCtx.getHelpID();
+        LOGGER.log(Level.INFO, "display '{0}' from {1}", new Object[]{helpId, helpSource});
```

Comment:
remove the single quotes from `'{0}'` as the quotes are actually discarding format specifier (according to sonar)

Reply:
thank you - fixed.